transport/http2_client.go

```diff
@@ -318,6 +329,9 @@ func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (_ *Strea
 func (t *http2Client) CloseStream(s *Stream, err error) {
 	t.mu.Lock()
 	delete(t.activeStreams, s.id)
+	if t.streamsQuota != nil {
+		t.streamsQuota.add(1)
```

Comment:
yes, it is atomic and does not matter. But I agree that it is unpleasant to make this inconsistent. I did not do that because it will introduce another a few booleans to this code.  Let me do it now.